### PR TITLE
Add PLDs to several Gaelco games [caius]

### PR DESCRIPTION
**Pull request by clawgrip**

### CODE REVIEW

File: src/mame/drivers/gaelco.cpp [hunk 1]
```diff
@@ -1089,49 +1093,61 @@ REF.922804/1
...
+	ROM_REGION( 0x80000, "oki", 0 )    // ADPCM samples - sound chip is OKIM6295
 	ROM_LOAD( "squash.d01", 0x000000, 0x80000, CRC(a1b9651b) SHA1(a396ba94889f70ea06d6330e3606b0f2497ff6ce) )
-	/* 0x00000-0x2ffff is fixed, 0x30000-0x3ffff is bank switched from all the ROMs */
+	// 0x00000-0x2ffff is fixed, 0x30000-0x3ffff is bank switched from all the ROMs
+
+	ROM_REGION( 0x4dc, "plds", 0)
+        ROM_LOAD ( "squashv1_gal16v8.f2",  0x000, 0x117, CRC(d5ed5985) SHA1(a4e9c8e3a7774e2a02fbca3ddf8175cf251825ba) ) // Unprotected
+        ROM_LOAD ( "squashv1_gal16v8.j16", 0x117, 0x117, CRC(fe78b903) SHA1(c806e63ce56a77f631043c184a42bf77ebda8a09) ) // Unprotected
+        ROM_LOAD ( "squashv1_gal20v8.d21", 0x22e, 0x157, CRC(a715e392) SHA1(31ebc78b084d49cc2f6479cbd42738e6bfbfb46a) ) // Unprotected
+        ROM_LOAD ( "squashv1_gal20v8.h11", 0x385, 0x157, CRC(51e34bc2) SHA1(381a898b3afb709e7d8e0f87df106f23aec2ccbe) ) // Unprotected
```
Comment (cuavas) [inline src/mame/drivers/gaelco.cpp:1115]: Please fix the indentation.  The rest of this file has one tab indentation for all ROM load commands.

File: src/mame/drivers/gaelco.cpp [hunk 2]
```diff
@@ -1089,49 +1093,61 @@ REF.922804/1
...
+	ROM_REGION( 0x100000, "oki", 0 )    // ADPCM samples - sound chip is OKIM6295
 	ROM_LOAD( "sound", 0x000000, 0x100000, CRC(99f80961) SHA1(de3a514a8f46dffd5f762e52aac1f4c3b08e2e18) )
-	/* 0x00000-0x2ffff is fixed, 0x30000-0x3ffff is bank switched from all the ROMs */
+	// 0x00000-0x2ffff is fixed, 0x30000-0x3ffff is bank switched from all the ROMs
+
+	ROM_REGION( 0x4dc, "plds", 0)
+        ROM_LOAD ( "thunderhoop_gal16v8.f2",  0x000, 0x117, BAD_DUMP CRC(d5ed5985) SHA1(a4e9c8e3a7774e2a02fbca3ddf8175cf251825ba) ) // Bruteforced but verified
+        ROM_LOAD ( "thunderhoop_gal16v8.j16", 0x117, 0x117, BAD_DUMP CRC(fe78b903) SHA1(c806e63ce56a77f631043c184a42bf77ebda8a09) ) // Bruteforced but verified
+        ROM_LOAD ( "thunderhoop_gal20v8.d21", 0x22e, 0x157, BAD_DUMP CRC(a715e392) SHA1(31ebc78b084d49cc2f6479cbd42738e6bfbfb46a) ) // Bruteforced but verified
+        ROM_LOAD ( "thunderhoop_gal20v8.h11", 0x385, 0x157, BAD_DUMP CRC(51e34bc2) SHA1(381a898b3afb709e7d8e0f87df106f23aec2ccbe) ) // Bruteforced but verified
```
Comment (cuavas) [inline src/mame/drivers/gaelco.cpp:1150]: Please fix the indentation here, too.

File: src/mame/drivers/gaelco2.cpp [hunk 3]
```diff
@@ -1195,57 +1195,63 @@ JP3  - 4 pin light gun header (player 2)
...
+	ROM_LOAD( "bang20.ic20", 0x0600000, 0x0080000, CRC(a1145df8) SHA1(305cda041a6f201cb011982f1bf1fc6a4153a669) )   // GFX only
+	ROM_LOAD( "bang13.ic13", 0x0680000, 0x0080000, CRC(fe3e8d07) SHA1(7a37561b1cf422b47cddb8751a6b6d57dec8baae) )   // GFX only
+	ROM_LOAD( "bang5.ic5",   0x0700000, 0x0080000, CRC(9bee444c) SHA1(aebaa3306e7e5aada99ed469da9bf64507808cff) )   // Sound only
+	ROM_FILL(                0x0780000, 0x0080000, 0x00 )            // Empty
+	ROM_LOAD( "bang21.ic21", 0x0800000, 0x0080000, CRC(fd93d7f2) SHA1(ff9d8eb5ac8d9757132aa6d79d2f7662c14cd650) )   // GFX only
+	ROM_LOAD( "bang14.ic14", 0x0880000, 0x0080000, CRC(858fcbf9) SHA1(1e67431c8775666f4839bdc427fabf59ffc708c0) )   // GFX only
+	ROM_FILL(                0x0900000, 0x0100000, 0x00 )            // Empty
+
+	ROM_REGION( 0x400, "plds", 0)
+    ROM_LOAD ( "bang_gal16v8.ic56", 0x000, 0x117, BAD_DUMP CRC(226923ac) SHA1(b1cac5208673183f401702ba844e1016d5fa4ea0) ) // Bruteforced but verified
```
Comment (cuavas) [inline src/mame/drivers/gaelco2.cpp:1224]: Please fix the indentation here, too.

File: src/mame/drivers/gaelco2.cpp [hunk 4]
```diff
@@ -1195,57 +1195,63 @@ JP3  - 4 pin light gun header (player 2)
...
 	ROM_LOAD( "bang5.ic5",   0x0700000, 0x0080000, CRC(9bee444c) SHA1(aebaa3306e7e5aada99ed469da9bf64507808cff) )   /* Sound only */
-	ROM_FILL(                0x0780000, 0x0080000, 0x00 )            /* Empty */
+	ROM_FILL(                0x0780000, 0x0080000, 0x00 )            // Empty
 	ROM_LOAD( "bang-a.ic21", 0x0800000, 0x0080000, CRC(531ce3b6) SHA1(196bb720591acc082f815b609a7cf1609510c8c1) )   /* GFX only */
 	ROM_LOAD( "bang-a.ic14", 0x0880000, 0x0080000, CRC(f8e1cf84) SHA1(559c08584094e605635c5ef3a25534ea0bcfa199) )   /* GFX only */
-	ROM_FILL(                0x0900000, 0x0100000, 0x00 )            /* Empty */
+	ROM_FILL(                0x0900000, 0x0100000, 0x00 )            // Empty
+
+	ROM_REGION( 0x117, "plds", 0)
+    ROM_LOAD ( "bang_gal16v8.ic56", 0x000, 0x117, BAD_DUMP CRC(226923ac) SHA1(b1cac5208673183f401702ba844e1016d5fa4ea0) ) // Bruteforced but verified
```
Comment (cuavas) [inline src/mame/drivers/gaelco2.cpp:1254]: The indentation is messed up here, too.

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/drivers/gaelco.cpp (hunk 1 above)
Comment (clawgrip) [reply src/mame/drivers/gaelco.cpp]: Fixed, thanks!

File: src/mame/drivers/gaelco2.cpp (hunk 3 above)
Comment (clawgrip) [reply src/mame/drivers/gaelco2.cpp]: Fixed, thanks!

File: src/mame/drivers/gaelco2.cpp (hunk 4 above)
Comment (clawgrip) [reply src/mame/drivers/gaelco2.cpp]: Fixed, thanks!

File: src/mame/drivers/gaelco.cpp (hunk 2 above)
Comment (clawgrip) [reply src/mame/drivers/gaelco.cpp]: Fixed, thanks!

**Pull request merged**